Sources/PackageRegistry/RegistryClient.swift

```diff
@@ -465,6 +569,82 @@ private extension RegistryManager {
     }
 }
 
+// MARK: - Serialization
+
+extension RegistryClient {
+    public enum Serialization {
+
+        public struct PackageMetadata: Codable {
+            public var releases: [String: Release]
+
+            public init(releases: [String: Release]) {
+                self.releases = releases
+            }
+
+            public struct Release: Codable {
+                public var url: String
+                public var problem: Problem?
+
+                public init(url: String, problem: Problem? = .none) {
+                    self.url = url
+                    self.problem = problem
+                }
+            }
+
+            public struct Problem: Codable {
+                public var status: Int
+                public var title: String
+                public var detail: String
+
+                public init(status: Int, title: String, detail: String) {
+                    self.status = status
+                    self.title = title
+                    self.detail = detail
+                }
+            }
+        }
+
+        public struct VersionMetadata: Codable {
+            public var id: String
+            public var version: String
+            public var resources: [Resource]
+            public var metadata: AdditionalMetadata
```

Comment:
Do we need this?

Reply:
I was assuming we would need this, see https://github.com/apple/swift-package-manager/pull/3873#discussion_r751942575